File: common/src/main/java/com/ssblur/scriptor/word/action/potions/StrengthAction.java
package com.ssblur.scriptor.word.action.potions;

import com.ssblur.scriptor.api.word.Descriptor;
import com.ssblur.scriptor.api.word.Word;
import com.ssblur.scriptor.helpers.targetable.Targetable;
import net.minecraft.world.effect.MobEffects;
import net.minecraft.world.item.BoneMealItem;
import net.minecraft.world.item.ItemStack;

public class StrengthAction extends PotionAction {
  public StrengthAction() {
    super(MobEffects.DAMAGE_BOOST, 40, 1d/3d, new Word.Cost(9, Word.COSTTYPE.ADDITIVE));
  }

  @Override
  public void applyToPosition(Targetable caster, Targetable targetable, Descriptor[] descriptors, double strength, double duration) {
    boolean playBoneMealAnimation = false;

    var level = targetable.getLevel();
    var pos = targetable.getOffsetBlockPos();

    for(int i = 0; i < strength + 1; i++)
      if(
        BoneMealItem.growCrop(ItemStack.EMPTY, level, pos)
          || BoneMealItem.growWaterPlant(ItemStack.EMPTY, level, pos, targetable.getFacing().getOpposite())
      )
        playBoneMealAnimation = true;

    if(playBoneMealAnimation) level.levelEvent(1505, pos, 0);
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/action/potions/StrengthAction.kt
package com.ssblur.scriptor.word.action.potions

import com.ssblur.scriptor.api.word.Descriptor
import com.ssblur.scriptor.helpers.targetable.Targetable
import net.minecraft.world.effect.MobEffects
import net.minecraft.world.item.BoneMealItem
import net.minecraft.world.item.ItemStack

class StrengthAction: PotionAction(MobEffects.DAMAGE_BOOST, 40.0, 1.0 / 3.0, Cost(9.0, COSTTYPE.ADDITIVE)) {
  override fun applyToPosition(
    caster: Targetable?,
    targetable: Targetable?,
    descriptors: Array<Descriptor>?,
    strength: Double,
    duration: Double
  ) {
    var playBoneMealAnimation = false

    val level = targetable!!.level
    val pos = targetable.offsetBlockPos

    var i = 0
    while (i < strength + 1) {
      if (BoneMealItem.growCrop(ItemStack.EMPTY, level, pos)
        || BoneMealItem.growWaterPlant(ItemStack.EMPTY, level, pos, targetable.facing.opposite)
      ) playBoneMealAnimation = true
      i++
    }

    if (playBoneMealAnimation) level.levelEvent(1505, pos, 0)
  }
}
